src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -449,6 +453,53 @@ - (BOOL)setupCommissioningSessionWithPayload:(MTRSetupPayload *)payload
     return [self syncRunOnWorkQueueWithBoolReturnValue:block error:error];
 }
 
+- (BOOL)setupCommissioningSessionWithDiscoveredDevice:(MTRCommissionableBrowserResult *)discoveredDevice
+                                              payload:(MTRSetupPayload *)payload
+                                            newNodeID:(NSNumber *)newNodeID
+                                                error:(NSError * __autoreleasing *)error
+{
+    auto block = ^BOOL {
+        chip::NodeId nodeId = [newNodeID unsignedLongLongValue];
+        self->_operationalCredentialsDelegate->SetDeviceID(nodeId);
+
+        auto errorCode = CHIP_ERROR_INVALID_ARGUMENT;
+        if (discoveredDevice.params.HasValue()) {
+            auto params = discoveredDevice.params.Value();
```

Comment:
The following lines are affecting calling `params.SetSetupPINCode`. It  is unclear to me that we really want to change the original input.

Reply:
What might make more sense, then, is to make a copy of discoveredDevice.params on the stack to start with.  In particular, I suspect that every get of discoveredDevice.params in fact does a copy...  But it would be better to make that explicit, maybe even by clearly writing out the type without `auto`.